common/utilities/number/stabilized-value.h

```diff
@@ -0,0 +1,83 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+
+namespace utilities {
+namespace number {
+
+    // The stabilized value implement a history based stable value. 
+    // It supply a way to define and get values with a required stabilization.
+    // When the user adds a value, the value is inserted into the history
+    // When the user call "get" the last stable value is returned
+    // Inputs: Template <value_type>
+    //         required history size
+    //         required stabilization percentage
+    template<typename T>
+    class stabilized_value
+    {
+    public:
+        stabilized_value( size_t history_size, float stabilize_percentage )
+            : _history_size( history_size )
+            , _last_stable_value( 0 )
+            , _stabilize_percentage( stabilize_percentage )
+        {
+            if( ( stabilize_percentage <= 0.0f ) || ( stabilize_percentage > 1.0f ) )
+                throw std::runtime_error( "Illegal value for stabilize_percentage: " + std::to_string(stabilize_percentage));
+        }
+
+        void add(T val)
+        {
+            if (_values.empty())
+            {
+                _values.push_back(val);
+                _last_stable_value = val;
+                return;
+            }
+
+            _values.push_back(val);
+            if (_values.size() > _history_size)
+                _values.pop_front();
+
+            std::unordered_map<T, int> values_count_map;
```

Comment:
I'm wondering if we should do this part in `get()` and not in `add()` since add gets called many times for one get?

Reply:
Consider cache the value and move calculate counters on get

Reply:
Done